pkg/model/mselection/heaviest.go

```diff
@@ -191,23 +245,21 @@ func (s *HeaviestSelector) OnNewSolidBundle(bndl *tangle.Bundle) {
 					panic(fmt.Sprintf("transaction not found: %v", hornet.Hash(approveeHash).Trytes()))
 				}
 
-				_, approveeORTSI = s.getTransactionRootSnapshotIndexes(cachedApproveeTx.Retain(), lsmi) // tx +1
+				_, approveeORTSI = dag.GetTransactionRootSnapshotIndexes(cachedApproveeTx.Retain(), lsmi) // tx +1
```

Comment:
Maybe move this whole check to a separate function where you pass the aproveeHashes and the belowMaxDepth